pkg/mutation/path/parser/node.go

```diff
@@ -57,6 +66,19 @@ func (r Path) DeepCopy() Path {
 	return out
 }
 
+func (r Path) String() string {
+	result := strings.Builder{}
+	for i, n := range r.Nodes {
+		nStr := n.String()
+		if _, isObject := n.(*Object); i > 0 && isObject {
```

Comment:
nit: the Node interface includes a Type method for this purpose, so this could be written as:

```
if i > 0 && n.Type() == ObjectNode {
```